torch/_inductor/fx_passes/post_grad.py

```diff
@@ -47,6 +48,91 @@
 inference_patterns = PatternMatcherPass()
 
 
+def get_storage(t):
+    return t.untyped_storage()._cdata
+
+
+def get_node_storage(node):
+    if not isinstance(node.meta["val"], torch.Tensor):
+        return None
+    if not torch._C._has_storage(node.meta["val"]):
+        return None
+    return get_storage(node.meta["val"])
+
+
+class FakeTensorUpdater:
+    def __init__(self, graph):
+        self.processed_hashes = set()
+        self.graph = graph
+
+        for node in self.graph.nodes:
+            self.processed_hashes.add(self.hash_node(node))
+
+    def hash_node(self, node):
+        return (node, node.target, node.args, node.kwargs)
+
+    def incremental_update(self):
```

Comment:
Would be nice to have a version that takes in an initial set of nodes, and doesn't require processing the whole graph, potentially ignoring aliasing

Reply:
What does this mean?

Reply:
Like, let's say modify some number of nodes, but dont want to have to deal with the metas myself, so I run the updater for some [start, end] nodes in the graph.